wgpu-core/src/device/global.rs

```diff
@@ -0,0 +1,2574 @@
+        profiling::scope!("CommandEncoder::drop");
+        log::debug!(
+            "command encoder {:?} is asked to be dropped",
+            command_encoder_id
+        );
+
+        let hub = A::hub(self);
+
+        let cmd_buf = hub.command_buffers.unregister(command_encoder_id).unwrap();
+        cmd_buf
+            .device
+            .untrack(&cmd_buf.data.lock().as_ref().unwrap().trackers);
+    }
+
+    pub fn command_buffer_drop<A: HalApi>(&self, command_buffer_id: id::CommandBufferId) {
+        profiling::scope!("CommandBuffer::drop");
+        log::debug!(
+            "command buffer {:?} is asked to be dropped",
+            command_buffer_id
+        );
+        self.command_encoder_drop::<A>(command_buffer_id)
+    }
+
+    pub fn device_create_render_bundle_encoder(
+        &self,
+        device_id: DeviceId,
+        desc: &command::RenderBundleEncoderDescriptor,
+    ) -> (
+        id::RenderBundleEncoderId,
+        Option<command::CreateRenderBundleError>,
+    ) {
+        profiling::scope!("Device::create_render_bundle_encoder");
+        let (encoder, error) = match command::RenderBundleEncoder::new(desc, device_id, None) {
+            Ok(encoder) => (encoder, None),
+            Err(e) => (command::RenderBundleEncoder::dummy(device_id), Some(e)),
+        };
+        (Box::into_raw(Box::new(encoder)), error)
+    }
+
+    pub fn render_bundle_encoder_finish<A: HalApi>(
+        &self,
+        bundle_encoder: command::RenderBundleEncoder,
+        desc: &command::RenderBundleDescriptor,
+        id_in: Input<G, id::RenderBundleId>,
+    ) -> (id::RenderBundleId, Option<command::RenderBundleError>) {
+        profiling::scope!("RenderBundleEncoder::finish");
+
+        let hub = A::hub(self);
+
+        let fid = hub.render_bundles.prepare(id_in);
+
+        let error = loop {
+            let device = match hub.devices.get(bundle_encoder.parent()) {
+                Ok(device) => device,
+                Err(_) => break command::RenderBundleError::INVALID_DEVICE,
+            };
+            #[cfg(feature = "trace")]
+            if let Some(ref mut trace) = *device.trace.lock() {
+                trace.add(trace::Action::CreateRenderBundle {
+                    id: fid.id(),
+                    desc: trace::new_render_bundle_encoder_descriptor(
+                        desc.label.clone(),
+                        &bundle_encoder.context,
+                        bundle_encoder.is_depth_read_only,
+                        bundle_encoder.is_stencil_read_only,
+                    ),
+                    base: bundle_encoder.to_base_pass(),
+                });
+            }
+
+            let render_bundle = match bundle_encoder.finish(desc, &device, hub) {
+                Ok(bundle) => bundle,
+                Err(e) => break e,
+            };
+
+            let (id, resource) = fid.assign(render_bundle);
+            log::info!("Created RenderBundle {:?}", id);
```

Comment:
The "Created Foo" logging seems to be a mix of `log::info!` and `log::debug!`. Let's pick one and stick with it.